### DISCUSSION TITLE
Figure captions not inheriting master slide font settings

### DISCUSSION THREAD

**Discussion by yonisidi** [quarto-dev/quarto-cli#10052] [Q&A]
### Description

I have a figure in a slide that has a caption.

When it renders the caption it does not use the font defined in the master slide template (screen grab of Two Column Layout below). Do I need to add something to layout for the caption to have a place to go? currently it looks like the caption object is creating as a side-effect.

 
````qmd
## My Header

::: columns
::: {.column}
- @fig-myfig shows something amazing. 
- bullet 2
- bullet 3
:::
::: {.column}
```{r fig-myfig}
#| fig-cap: 'my caption'
knitr::include_graphics('my-figure.png')
```
:::
:::
````

**Comment by mcanouil** [comment]
Could you share a **small self-contained "working" (reproducible)** example to work with, _i.e._, a complete Quarto document or a Git repository? The goal is to make it as easy as possible for us to recreate your problem so that we can fix it: please help us help you! Thanks.

Did you follow the below documentations to create your template?
- https://quarto.org/docs/output-formats/ms-word-templates.html (this is for Word but the same principles apply and this page has screenshots)
- https://quarto.org/docs/presentations/powerpoint.html#powerpoint-templates
- https://pandoc.org/MANUAL.html#option--reference-doc

---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td></tr>
</table>

Additionally, and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Comment by cderv** [comment]
I believe there is another issue here, as caption will not show in output when `#fig-` prefixed label is used. 

Related to 
- https://github.com/quarto-dev/quarto-cli/issues/8667

**Comment by yonisidi** [comment]
captions come out ok for figures

The font isn't inherited correctly

_extension repo: https://github.com/yonisidi/cmei_quarto


```bash
quarto use template yonisidi/cmei_quarto
```

````qmd
---
title: "Slides Title"
author: "Author Name"
date: today
format: cmei-pptx
output-file: "slides.pptx"
---

```{r}
library(knitr)
```

# Background Section

## Goal of the Analysis

# Results Section

## Create Figure

::: columns
::: {.column width="50%"}

- @fig-test_create reference
- Bullet 2

:::

::: {.column width="50%"}
```{r fig-test_create}
#| fig-cap: 'Generating figure in chunk'
plot(1:10, 1:10)
```
:::
:::


## Read Figure

::: columns
::: {.column width="50%"}

- @fig-test_read reference
- Bullet 2

:::

::: {.column width="50%"}
```{r fig-test_read}
#| fig-cap: 'Reading from local storage'
knitr::include_graphics('figures/xkcd-image.png')
```
:::
:::


# Conclusions Section

## Conclusions 

## Bibliography


```

**Reply by cderv** [reply]
> captions come out ok for figures

Which version of Quarto are you using ? I am sure the bug it still here, and I get it when I run your example with latest pre-release 1.5 version. 

> The font isn't inherited correctly

Also to clarify: which is the expected font to be used and what is the actual font ? 

I think it should be check with Pandoc directly that Pandoc does the right thing as you expect. If Pandoc don't do it, we probably don't do it either. 

I mean really what is defining the figure caption font you expect in the pptx template passed as reference doc ?

From quick testing, I think it could be a Pandoc issue.

**Reply by yonisidi** [reply]
```bash
$ quarto --version
1.3.340
```

In the case of the expected font, I would think that if the master slide is set to font XXX, then any object created should inherit it and not the default powerpoint font (not even talking about the font size).

**Reply by mcanouil** [reply]
Note that the current version of Quarto is 1.4.555 and that 1.5.45 is in "release candidate" state.
I strongly suggest (if you can) to upgrade your Quarto version at least to 1.4.

**Reply by cderv** [reply]
> In the case of the expected font, I would think that if the master slide is set to font XXX, then any object created should inherit it and not the default powerpoint font (not even talking about the font size).

I do think this is a Pandoc's limitation. I don't know how they create the caption but directly using `pandoc` CLI with a reference document  gives me the same output. 

You should try to convert a .md doc with an image & caption, pass your reference doc and see about the caption font. It could be a Pandoc issue

**Reply by yonisidi** [reply]
@cderv you are right, direct to pandoc creates same problem:

```bash
pandoc test2.md -o test2.pptx --reference-doc=_extensions/yonisidi/cmei/resources/cmei_template.pptx
```

```md
---
title: Two Columns Slide with Image
author: Your Name
date: \today
header-includes:
  - \usepackage{multicol}
---

# Slide Title

::: columns

::: column
This is the content of the first column.
:::

:::column
![Image Caption](https://placehold.co/600x400.png)
:::

:::
```

**Reply by cderv** [reply]
Unfortunately, this means we need Pandoc to do better...

**Reply by yonicd** [reply]
I opened an issue on the pandoc repo https://github.com/jgm/pandoc/issues/9896

**Reply by yonisidi** [reply]
looks like this is an issue bouncing around in [SO](https://stackoverflow.com/questions/75077107/how-to-change-font-size-in-figure-caption-in-a-pptx-rendered-by-quarto-and-pando) too

**Comment by yonisidi** [comment]
Same issue with tables

The font settings of the table contents and caption of the table are not inherited from the master slide, you can see the difference between the font in the left column and the font in the right column

````qmd
---
title: "Slides Title"
author: "Author Name"
date: today
format: cmei-pptx
output-file: "slides.pptx"
---

```{r}
library(knitr)
```

# Background Section

## Goal of the Analysis

# Results Section

## Create Figure

::: columns
::: {.column width="50%"}

- @fig-test_create reference
- Bullet 2

:::

::: {.column width="50%"}
```{r fig-test_create}
#| fig-cap: 'Generating figure in chunk'
plot(1:10, 1:10)
```
:::
:::


## Read Figure

::: columns
::: {.column width="50%"}

- @fig-test_read reference
- Bullet 2

:::

::: {.column width="50%"}
```{r fig-test_read}
#| fig-cap: 'Reading from local storage'
knitr::include_graphics('figures/xkcd-image.png')
```
:::
:::

## Create Table

::: columns
::: {.column width="50%"}
| Default | Left | Right | Center |
|---------|:-----|------:|:------:|
| 12      | 12   |    12 |   12   |
| 123     | 123  |   123 |  123   |
| 1       | 1    |     1 |   1    |

: Demonstration of pipe table syntax
:::

::: {.column width="50%"}
some content that goes with the table

:::
:::


# Conclusions Section

## Conclusions 

## Bibliography


````
